### PR TITLE
fix: properly handle arguments for casts

### LINKED ISSUE #1525: Adding arguments to custom model casts via $casts[] results in incorrect property type
- Larastan Version: 2.4.0
- `--level` used: 5

### Description
When using `Castable` classes and providing arguments in the eloquent model's `$casts` definition, the property is not being casted to a `phpstan` type.

Model property casting has been handled in #1333.

### How to replicate

Implement a `castable` class as defined [here](https://laravel.com/docs/9.x/eloquent-mutators#castables) and provide an argument via the model's `$casts[]`.

```php
use App\Models\Address;
 
protected $casts = [
    'address' => Address::class.':argument',
];
```

### PR DESCRIPTION
**Pull request by erikgaal**
- [x] Added or updated tests
- [x] Documented user facing changes

**Changes**

Fixes #1525.

**Breaking changes**

### CODE REVIEW

File: src/Properties/ModelCastHelper.php [hunk 1]
```diff
@@ -158,9 +161,9 @@ public function getDateType(): Type
 
     /**
      * @param  string  $cast
-     * @return string|null
+     * @return string
```
Comment (szepeviktor) [inline src/Properties/ModelCastHelper.php:164]: For me this is like a dream-come-true.
Thank you!

**Comment by szepeviktor** [comment]
@canvural Seems like CI fails because of a reason outside this PR.
What do you think?

**Comment by szepeviktor** [comment]
@erikgaal Yes, these are not your problems.  #1530

**Comment by szepeviktor** [comment]
Now everything should be okay ✔️

**Review by szepeviktor — APPROVED** [review]

File: src/Properties/ModelCastHelper.php [hunk 2]
```diff
@@ -101,7 +102,9 @@ public function getWriteableType(string $cast, Type $originalType): Type
             'collection' => new ObjectType('Illuminate\Support\Collection'),
             'date', 'datetime' => $this->getDateType(),
             'immutable_date', 'immutable_datetime' => new ObjectType('Carbon\CarbonImmutable'),
-            'Illuminate\Database\Eloquent\Casts\AsArrayObject', 'Illuminate\Database\Eloquent\Casts\AsCollection' => new MixedType(),
+            'Illuminate\Database\Eloquent\Casts\AsArrayObject', 'Illuminate\Database\Eloquent\Casts\AsCollection',
+            'Illuminate\Database\Eloquent\Casts\AsEncryptedArrayObject', 'Illuminate\Database\Eloquent\Casts\AsEncryptedCollection' => new MixedType(),
+            'Illuminate\Database\Eloquent\Casts\AsStringable' => TypeCombinator::union(new StringType(), new ObjectType('Stringable')),
```
Comment (canvural) [inline src/Properties/ModelCastHelper.php:107]: Can it also accept `Illuminate\Support\Stringable`?

Comment (erikgaal) [reply src/Properties/ModelCastHelper.php:107]: Yes! Laravel's `Stringable` has a `__toString` method, so therefore it implicitly implements the native `Stringable` interface.

**Review by canvural — APPROVED** [review]

**Comment by canvural** [comment]
Thank you!

**Comment by rovansteen** [comment]
@canvural would it be possible to put out a new release that includes this fix? 🙏🏻

**Comment by szepeviktor** [comment]
@rovansteen Here we go! 🚲
https://github.com/nunomaduro/larastan/releases/tag/2.4.1
Take cover! 🙃

**Comment by rovansteen** [comment]
@szepeviktor thank you very much!

**Pull request merged**